client/src/app/sidebar/sidebar.component.ts

```diff
@@ -1,29 +1,128 @@
+
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
+  @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
+  
+  
+
+constructor(private appService: AppService, private dialog: MatDialog, private fb: FormBuilder) {
     const filter = appService.filterValue();
     this.checkedRegions = new Set(filter.regions);
     this.checkedSectors = new Set(filter.sectors);
+    this.masterSelectedRegion = true;
+    this.masterSelectedSector = true;
+    this.checklistRegion = [
+      {id: 1, value: 'North America', isSelected: true},
+      {id: 2, value: 'Mexico / Central America', isSelected: true},
+      {id: 3, value: 'South America', isSelected: true},
+      {id: 4, value: 'Europe', isSelected: true},
+      {id: 5, value: 'Middle East / North Africa', isSelected: true},
+      {id: 6, value: 'Sub-Saharan Africa', isSelected: true},
+      {id: 7, value: 'China', isSelected: true},
+      {id: 8, value: 'India', isSelected: true},
+      {id: 9, value: 'Other', isSelected: true},
+    ]
+    this.checklistSector = [
+      {id: 1, value: 'Agriculture / Food Security', isSelected: true},
+      {id: 2, value: 'Collective Mobilization / Advocacy', isSelected: true},
+      {id: 3, value: 'Disaster / Humanitarian Relief', isSelected: true},
+      {id: 4, value: 'Education', isSelected: true},
+      {id: 5, value: 'Energy', isSelected: true},
+      {id: 6, value: 'Environment / Sustainability', isSelected: true},
+      {id: 7, value: 'Fair Trade / Market Access', isSelected: true},
+      {id: 8, value: 'Gender Empowerment', isSelected: true},
+      {id: 9, value: 'Health', isSelected: true},
+      {id: 10, value: 'Housing', isSelected: true},
+      {id: 11, value: 'Human Rights / Law', isSelected: true},
+      {id: 12, value: 'Hunger / Malnutrition', isSelected: true},
+      {id: 13, value: 'Law', isSelected: true},
+      {id: 14, value: 'Immigration', isSelected: true},
+      {id: 15, value: 'Information Technology', isSelected: true},
+      {id: 16, value: 'Microfinance', isSelected: true},
+      {id: 17, value: 'Refugee / Displaced Persons', isSelected: true},
+      {id: 18, value: 'Water / Sanitation', isSelected: true},
+      {id: 19, value: 'Other', isSelected: true},
+    ]
+  }
+  
+  checkUncheckAllRegion() {
+    this.masterSelectedRegion = !this.masterSelectedRegion;
+    for (var i = 0; i < this.checklistRegion.length; i++) {
+      this.checklistRegion[i].isSelected = this.masterSelectedRegion;
+    }
+    this.onRegionChange();
+
+  }
+  
+
+  isAllSelectedRegion(id: number){
+    var totalSelected = 0;
+    for(var i = 0; i < this.checklistRegion.length; i++){
+      if(this.checklistRegion[i].id == id){
+        this.checklistRegion[i].isSelected = !this.checklistRegion[i].isSelected;
+      }
+      if(this.checklistRegion[i].isSelected == true){
+        totalSelected = totalSelected + 1;
+      }
+    }
+    this.masterSelectedRegion = (totalSelected == 9);
```

Comment:
i would avoid using constant values for comparison `this.masterSelectedRegion = (totalSelected == 9);`, instead use `totalSelected == this.checklistRegion.length` to make it easier to maintain in future